shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/communication/DatabaseCommunicationEngine.java

```diff
@@ -100,6 +103,20 @@ public DatabaseCommunicationEngine(final String driverType, final ShardingSphere
     
     protected void refreshMetaData(final ExecutionContext executionContext) throws SQLException {
         SQLStatement sqlStatement = executionContext.getSqlStatementContext().getSqlStatement();
+        Boolean needToRefreshMetaData = NEED_TO_REFRESH_META_DATA.get(sqlStatement.getClass());
+        if (null == needToRefreshMetaData) {
+            needToRefreshMetaData = NEED_TO_REFRESH_META_DATA.computeIfAbsent(sqlStatement.getClass(),
```

Comment:
This is for refresh only, could we consider about move them into MetaDataRefreshEngine class?

Reply:
I think so.